### PR TITLE
Feat (tables): implements tan stack table

### PR DESCRIPTION
**Pull request by Francisco-Villanueva**

### CODE REVIEW

File: src/app/home/my-stock/DataStock.tsx [hunk 1]
```diff
@@ -1,48 +1,210 @@
...
+        accessorFn: (row) => row.serialNumber,
+        header: "Serial",
+        size: 200,
+        cell: ({ getValue }) => <span>{getValue<string>()}</span>,
+      },
+  {
+    id: "expander",
+    header: () => null,
+    size: 10,
+    cell: ({ row }) => {
```
Comment (sebikap-gm2) [inline src/app/home/my-stock/DataStock.tsx:95]:
```
interface ExpanderCellProps {
   row: TableRow<unknown>
}
const ExpanderCell = ({ row }) => (
   ...
)
```

File: src/app/home/my-team/page.tsx [hunk 2]
```diff
@@ -3,11 +3,18 @@ import { PageLayout } from "@/common";
...
-
+import { useEffect } from "react";
+import { Memberservices } from "@/services";
 export default function MyTeam() {
   const {
-    members: { members },
+    members: { members, setMembers },
   } = useStore();
+
+  useEffect(() => {
```
Comment (sebikap-gm2) [inline src/app/home/my-team/page.tsx:13]: Remove

File: src/app/home/orders/EquipmentTable.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    accessorKey: "status",
+    header: "Status",
+    cell: ({ getValue }) => <OrderState status={getValue<OrderStatus>()} />,
+  },
+  {
+    accessorKey: "products",
+    header: "Total",
+    cell: ({ getValue }) => (
+      <span>
+        USD {getValue<Product[]>().reduce((a, b) => parseInt(b.price) + a, 0)}
```
Comment (sebikap-gm2) [inline src/app/home/orders/EquipmentTable.tsx:45]: Find out if we can avoid passing types and let ts infer them

File: src/app/home/shipments/DataShipments.tsx [hunk 4]
```diff
@@ -1,10 +1,106 @@
-import { Layout } from "@/common";
-import { TableShipments } from "@/components";
+"use client";
+import { ArrowRight, Button, CustomLink } from "@/common";
+import { Table } from "@/components";
+import { useStore } from "@/models";
+import { Product, ProductTable, Shipment, ShipmentTable } from "@/types";
+import { ColumnDef } from "@tanstack/react-table";
+import { format } from "date-fns";
```
Comment (sebikap-gm2) [inline src/app/home/shipments/DataShipments.tsx:7]: God please no

File: src/app/home/shipments/DataShipments.tsx [hunk 5]
```diff
@@ -1,10 +1,106 @@
...
+    id: "expander",
+    header: () => null,
+
+    size: 2,
+    cell: ({ row }) => {
+      return (
+        row.getCanExpand() && (
+          <div
+            className=" flex   "
+            onClick={() => handleSelect(row.original.orderId)}
```
Comment (sebikap-gm2) [inline src/app/home/shipments/DataShipments.tsx:65]: @Francisco-Villanueva Add meeting to review this

File: src/app/home/my-stock/DataStock.tsx [hunk 6]
```diff
@@ -1,48 +1,210 @@
...
+            }}
+          />
         </div>
-      </div>
+      </aside>
 
-      <div className="mt-10">
-        <Table />
+      <div className="h-[90%] top-[8%] w-full overflow-y-auto  absolute ">
+        <Table<Product>
```
Comment (sebikap-gm2) [inline src/app/home/my-stock/DataStock.tsx:199]:
```
interface ProductTableProps {
    products: Product[]
    internalProducts: Product[]
}

export const ProductTable = ({ products, internalProducts }: ProductTableProps) => {
   return (
      <Table<Product>
         data={products}
         columns={prodcutColumns({ serial: false })}
         getRowCanExpand={() => true}
         subComponent={internalProducts ? 
            <Table<Product> data={internalProducts} columns={InternalProductsColumns} /> : null
         }
      />
   )
}
```

File: src/app/home/shipments/ShipmentDetailsRow.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import { Product } from "@/types";
+import { observer } from "mobx-react-lite";
+import { prodcutColumns } from "../my-stock/DataStock";
+
+export default observer(function ShipmentDetailsRow() {
+  const {
+    shipments: { selectedShipment, setSelectedShipmentId, selectedShipmentId },
+  } = useStore();
+
+  return (
```
Comment (sebikap-gm2) [inline src/app/home/shipments/ShipmentDetailsRow.tsx:14]: Move to ProductTable component

File: src/app/home/shipments/DataShipments.tsx [hunk 8]
```diff
@@ -1,10 +1,106 @@
...
+    setSelectedShipmentId(shipmentId);
 
-export default function DataShipments() {
   return (
-    <Layout>
-      <TableShipments />
-    </Layout>
+    <div className="overflow-y-auto h-full w-full relative  ">
+      <div className="absolute w-full h-full ">
+        <Table<ShipmentTable>
```
Comment (sebikap-gm2) [inline src/app/home/shipments/DataShipments.tsx:97]: `<ProductTable products={selectedShipment.products} />`

File: src/common/DropFilter.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+"use client";
```
Comment (sebikap-gm2) [inline src/common/DropFilter.tsx:1]: Remove

File: src/common/TeamCard.tsx [hunk 10]
```diff
@@ -11,9 +11,11 @@ export function TeamCard({ team, className }: TeamCardProps) {
   const colors_role = {
```
Comment (sebikap-gm2) [inline src/common/TeamCard.tsx:11]: Remove

File: src/components/Navbar.tsx [hunk 11]
```diff
@@ -3,30 +3,39 @@ import React from "react";
...
-import { useRouter } from "next/navigation";
+import { ShopIcon } from "@/common/Icons";
+import { usePathname, useRouter } from "next/navigation";
 
 type NavbarProps = {
   title?: string;
   searchInput?: string;
   placeholder?: string;
-  hasNotification?: boolean;
 };
```
Comment (sebikap-gm2) [inline src/components/Navbar.tsx:14]:
```
const Titles = {
   'my-stock': 'My Stock',
   'my-team': 'My Team'
} as const


<h1>{Titles[pathname] ?? ''}</h1>
```

File: src/models/products.store.ts [hunk 12]
```diff
@@ -1,12 +1,28 @@
...
+    get productsTable(): ProductTable[] {
+      return store.products.map((product) => ({
+        category: {
+          category: product.category,
+          img: product.imgUrl,
+        },
+        quantity: product.stock,
+        model: {
+          model: product.model,
+          processor: product.processor || undefined,
```
Comment (sebikap-gm2) [inline src/models/products.store.ts:19]: No need for `|| undefined`

**Review by sebikap-gm2 — CHANGES_REQUESTED** [review]

File: src/app/home/orders/EquipmentTable.tsx (hunk 3 above)
Comment (Francisco-Villanueva) [reply src/app/home/orders/EquipmentTable.tsx:45]: image

File: src/components/Navbar.tsx [hunk 13]
```diff
@@ -3,30 +3,39 @@ import React from "react";
...
 export const Navbar = function ({
   title,
   searchInput,
   placeholder,
-  hasNotification,
 }: NavbarProps) {
   const router = useRouter();
+  const pathName = usePathname();
+
+  const Titles = {
```
Comment (sebikap-gm2) [inline src/components/Navbar.tsx:23]: Outside component

File: src/utils/date.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+  const day = date.getUTCDate();
+  const month = date.getUTCMonth() + 1; // Adding 1 because months are zero-indexed
+  const year = date.getUTCFullYear();
+
+  // Add leading zeros if necessary
+  const day2 = day < 10 ? "0" + day : day;
+  const month2 = month < 10 ? "0" + month : month;
+
+  // Format the date as DD/MM/YYYY
+  return day2 + "/" + month2 + "/" + year;
```
Comment (sebikap-gm2) [inline src/utils/date.ts:16]: `new Date().toLocaleDateString('en')`

File: src/components/Tables/ProductsTable.tsx [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+      <span className="text-sm">{getValue<string>()}</span>
+    ),
+  },
+  {
+    accessorFn: (row) => row.name,
+    header: "Currently with",
+    cell: ({ row }) => (
+      <span className="text-md">
+        {/* // TODO: nombre de la persona a la que se dirige el producto */}
+        Name & LastName
```
Comment (sebikap-gm2) [inline src/components/Tables/ProductsTable.tsx:131]: Add backend endpoint getAll that queries assigned and unassigned products.
**@Francisco-Villanueva THIS IS A TODO FOR A NEXT PR EHHHHHH**

**Review by sebikap-gm2 — CHANGES_REQUESTED** [review]

**Pull request merged**